### PR TITLE
pm: s/power-domain/power-domains and add power-domain-names

### PR DESCRIPTION
**Pull request by gmarull**
Some devices may belong to >1 power domain, so with the current design this is something not possible to describe. It's worth to note that Linux also uses the `power-domains` naming scheme, not `power-domain`. This patch also introduces `power-domain-names` so that each entry in `power-domains` can be given a name if needed.

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]

**Comment by dbaluta** [comment]
@gmarull  does Zephyr support multiple power domains attached to a device? Or this is just a name change for a property and the support for multiple power domain is in progress?

**Comment by gmarull** [comment]
> @gmarull does Zephyr support multiple power domains attached to a device? Or this is just a name change for a property and the support for multiple power domain is in progress?

Zephyr does not support them, but I need to describe peripherals that belong to >1 in an upcoming PR.

File: dts/bindings/base/pm.yaml [hunk 1]
```diff
@@ -16,15 +16,6 @@ properties:
       Wake up capable devices are disabled (interruptions will not wake up
       the system) by default but they can be enabled at runtime if necessary.
 
-  power-domain:
-    type: phandle
-    description: |
-
-      Power domain the device belongs to.
-
-      The device will be notified when the power domain it belongs to is either
```
Comment (JordanYates) [inline dts/bindings/base/pm.yaml:25]: Should this comment be migrated across?

Comment (gmarull) [reply dts/bindings/base/pm.yaml:25]: This is an implementation detail, not related to DT. DT just describes the domain connections, not how they're handled.

**Review by dbaluta — DISMISSED** [review]

**Review by LaurentiuM1234 — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by ceolin — APPROVED** [review]

File: doc/services/pm/power_domain.rst
Comment (JarmouniA) [inline doc/services/pm/power_domain.rst:1]: Forgot to update https://github.com/zephyrproject-rtos/zephyr/blob/6f64f77557967a41ef7df429f0a831787bb8418f/doc/services/pm/power_domain.rst?plain=1#L126

**Pull request merged**